code/player.dm

```diff
@@ -71,6 +86,23 @@
 		else
 			return src.rounds_seen
 
+	/// sets the join time to the current server time, in 1/10ths of a second
+	proc/log_join_time()
+		src.round_join_time = TIME
+
+	/// sets the leave time to the current server time, in 1/10ths of a second
+	proc/log_leave_time()
+		src.round_leave_time = TIME
```

Comment:
I'd argue that the `round_leave_time` var is unnecessary because this is the only place where you set it and you also immediately reset it to null right after that. You could have `calculate_player_time` just use `TIME` instead of this. But it's just a minor thing.